helloworld/elementary_4.c

```diff
@@ -0,0 +1,21 @@
+#include<stdio.h>
+
+int main() {
+
+    int szam;
+	int osszeg;
+  
+  	setbuf(stdout, NULL);
+	
+	printf("Írj egy számot: ");    
+	scanf("%d", &szam);
+
+	osszeg = (1+szam)*szam/2; // persze lehetett volna nagyon idétlen ciklussal is megoldani, de ez elegánsabb megoldásnak tűnt :-)
```

Comment:
👍 Ez felvet egy izgalmas témát az algoritmusokról, amivel még találkozni fogsz. Van az ún. "big O notation", ami azt fejezi ki, hogy egy algoritmus által igénybevett idő miként függ a bemenetek számától, méretétől, stb.

Ha kétszer annyi elemmel valami kétszer annyi idejig is tart, magyarán az idő n-nel egyenesen arányok, akkor az algoritmusra azt mondják, hogy **O(n)** komplexitású. Ha a szükséges idő négyzetesen nő a bemenettel, akkor az **O(n²)**. Ha úgy nő, mint a faktoriális, akkor **O(n!)**.

Itt ha ciklust használtál volna, akkor **O(n)** komplexitású algoritmust írtál volna, mert minél nagyobb számot ír be az ember, annál többször kell lefutnia a ciklusnak. Ez a megoldás viszont **O(1)** komplexitású, aminek örülünk, mert ugyanolyan gyors bármely számra.